client/src/components/FieldButton.tsx

```diff
@@ -48,14 +55,60 @@ function FieldButton({
     count: MatchScores;
     alliance: boolean | undefined;
 }) {
-    // const [alliance, setAlliance] = useState(false); // false=red, true=blue, null=hollow purple
+    const [pickupLocation, setPickupLocation] = useState<
+        PickupLocation | undefined
+    >();
 
     const handleCount = (autoKey: countKeys, teleKey: countKeys) => {
-        const finalKey = teleOp ? teleKey : autoKey;
-        setCount(prevCount => ({
-            ...prevCount,
-            [finalKey]: prevCount[finalKey] + 1,
-        }));
+        if (pickupLocation) {
+            const finalKey = teleOp ? teleKey : autoKey;
+            setCount(prevCount => ({
+                ...prevCount,
+                [finalKey]: prevCount[finalKey] + 1,
+            }));
+            const pickupKeys = {
+                preload: 'autoPreload',
+                pickup: 'autoPickup',
+                speaker: 'telePickupSpeaker',
+                middle: 'telePickupMiddle',
+                source: 'telePickupSource',
+            } as const;
+            setCount(prevCount => ({
+                ...prevCount,
+                [pickupKeys[pickupLocation]]:
+                    prevCount[pickupKeys[pickupLocation]] + 1,
+            }));
+        }
+        else if (
+            !(count.teleShootNear ||
+            count.teleShootMid ||
+            count.teleShootFar ||
+            count.teleAmp ||
+            count.teleMiss) &&
```

Comment:
What are these conditions for? it looks like it checks if any of these fields are nonzero?

Reply:
Oh is that because only first teleop after a hold should be counted as hold?